### PR TITLE
Distance and history

### PR DESCRIPTION
**Pull request by bt-s**

### CODE REVIEW

File: src/components/Books/Address.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  </div>
+);
+
+Address.propTypes = {
+  distance: PropTypes.string,
+  separator: PropTypes.object,
+  name: PropTypes.string
+};
+
+export default Address;
```
Comment (fadhilmch) [inline src/components/Books/Address.js:18]: What is this component for? I didn't see any other components calling this component

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -28,12 +33,18 @@ const SearchBase = props => {
...
 
     props.history.push(ROUTES.BOOKS);
   };
 
+  if (_.isEmpty(props.books) && props.hasSearched === false) {
+    navigator.geolocation.getCurrentPosition(pos => {
+      props.storeCoords({lat: pos.coords.latitude, lng: pos.coords.longitude});
+    });
+  }
+
```
Comment (fadhilmch) [inline src/components/Search/Search.js]: Do you think that we need to call user location once on the beginning and store it on redux?

Comment (fadhilmch) [reply src/components/Search/Search.js]: oh, you already store it right, so I will call the redux state on add book page

File: src/pages/BooksPage.js [hunk 3]
```diff
@@ -1,78 +1,92 @@
...
-    getBooks();
+  if (
+    _.isEmpty(props.books) &&
+    props.hasSearched === false &&
+    JSON.stringify(props.coords) !== JSON.stringify(initialCoords)
+  ) {
+    navigator.geolocation.getCurrentPosition(pos => {
+      props.storeCoords({lat: pos.coords.latitude, lng: pos.coords.longitude});
+      getBooks();
+    });
```
Comment (fadhilmch) [inline src/pages/BooksPage.js]: why don't you get the location from the redux state?

File: src/pages/HistoryPage.js [hunk 4]
```diff
@@ -10,32 +10,88 @@ const HistoryPage = props => {
...
+    console.log('////', props);
+
+    props.firebase.myUID &&
+      props.firebase
+        .user(props.firebase.getMyUID())
+        .get()
+        .then(user => {
+          console.log('user12///', user.data());
+          Promise.all(
+            user.data().transactions.map(id =>
```
Comment (fadhilmch) [inline src/pages/HistoryPage.js:22]: I got error on this line. 

"Unhandled Rejection (TypeError): Cannot read property 'map' of undefined"

**Review by fadhilmch — COMMENTED** [review]
Issue: when I go to the home, and then search for books and then go to the book details, it doesn't get the book ID (the url show 'undefined' ) and doesn't show anything on the book detail

I think it is better to not make the 'Request' Botton on the book details page floating in fixed position when we scroll the page.

**Review by fadhilmch — APPROVED** [review]

File: src/components/Books/Address.js (hunk 1 above)
Comment (bt-s) [reply src/components/Books/Address.js:18]: @Envl said that he will use this later.

File: src/components/Search/Search.js (hunk 2 above)
Comment (bt-s) [reply src/components/Search/Search.js]: Done!

File: src/pages/BooksPage.js (hunk 3 above)
Comment (bt-s) [reply src/pages/BooksPage.js]: Done!

File: src/pages/HistoryPage.js (hunk 4 above)
Comment (bt-s) [reply src/pages/HistoryPage.js:22]: I will investigate this.

**Pull request merged**